### DISCUSSION TITLE
How to add URL style QR code in Part page?

### DISCUSSION THREAD

**Discussion by wafflecrate** [inventree/InvenTree#4849] [Q&A] [answered]
Hi everyone-

New Inventree user looking for some guidance. I have a new Part entry and I want to generate a QR code that, when scanned on a smartphone brings the user to Inventory count page.

Because right now when I Click "Part Label", the label looks good, but the QR code links a generic string called {"part: 1} and attempts to open it in Google on my smartphone. 

[This article](https://docs.inventree.org/en/latest/report/labels/#url-style-qr-code) says you have to add the below HTML tag in order to make the QR code link back to the stock item page.

Which template file am I supposed to add that HTML tag into? I tried adding it into the qr.html file but then it made the QR code gigantic!

**Comment by SchrodingersGat** [comment, marked as answer]
@wafflecrate the QR code by itself is not meant to provide a fully-qualified URL. It is meant for scanning using a system which understands the context of the data.

For example the InvenTree mobile app can scan these barcodes natively. And if you are using a USB barcode scanner in the web interface, it will work too.

But scanning via a generic barcode scanner on your phone is not the intended use case.

**Reply by wafflecrate** [reply]
I understand now and you are correct. I was not using the QR code correctly (was just using my smartphone's camera app). Thank you for clarifying this. This issue has been solved.
